packages/suite/src/components/suite/BioAuthGuard/BioAuthGuard.tsx

```diff
@@ -120,29 +91,34 @@ const BioAuthOverlay = ({
 };
 
 export const BioAuthGuard = ({ children }: { children: React.ReactNode }) => {
+    const [isWindowFocused, setIsWindowFocused] = useState(true);
     const isBioAuthValidationRequired = useSelector(state =>
         selectIsBioAuthValidationRequired(state, new Date()),
     );
-    const isBioAuthAvailable = useSelector(selectBioAuthEnabled);
-    const isAppUiHidden = false; // NOTE: temporary
-    const dispatch = useDispatch();
+    const isBioAuthEnabled = useSelector(selectBioAuthEnabled);
 
-    useEffect(() => {
-        dispatch(checkBioAuthAvailableThunk());
-    }, [dispatch]);
+    const { isBioAuthAvailable, requestBioAuthValidation } = useBioAuthDesktopApi();
+
+    const dispatch = useDispatch();
 
     useEffect(() => {
-        if (!isBioAuthAvailable) {
-            return;
-        }
-        dispatch(bioAuthActions.initBioAuth(performance.now()));
+        // user never decided to use this feature -> no need to check whether bioAuth is available on the system level
+        if (!isBioAuthEnabled) return;
+        if (!isBioAuthAvailable) return;
 
+        let timeoutId: TimerId;
         const handleBlur = () => {
-            dispatch(bioAuthWindowBlurThunk(new Date()));
+            setIsWindowFocused(false);
+            dispatch(bioAuthActions.bioAuthWindowBlur(Date.now()));
+            timeoutId = setTimeout(() => {
```

Comment:
Shouldn't this business logic be clearly in the thunks?

Reply:
Originally, I was thinking of something like process time, ticks or something, to make the time check harder to cheat. But i couldn't find something more reliable. So I defaulted to teh timeout and time difference.